.github/actions/check-changes/action.yaml

```diff
@@ -31,26 +35,42 @@ runs:
       env:
         GIT_BASE_REF: ${{ inputs.baseRef }}
         GIT_REF: ${{ inputs.ref }}
+        GITHUB_EVENT_FORCED: ${{ github.event.forced }}
+        GITHUB_REF_BEFORE: ${{ github.event.before }}
         USE_FALLBACK: ${{ inputs.useFallback }}
+        DIFF_FILTER: ${{ inputs.diffFilter }}
       run: |
         : Check for Changed Files ✅
         if [[ "${RUNNER_DEBUG}" ]]; then set -x; fi
         shopt -s extglob
         shopt -s dotglob
 
-        if ! git cat-file -e "${GIT_BASE_REF}" &> /dev/null; then
-          echo "::warning::Provided base reference ${GIT_BASE_REF} is invalid"
-          if [[ "${USE_FALLBACK}" == 'true' ]]; then
-            GIT_BASE_REF='HEAD~1'
+        if [[ "${GIT_BASE_REF}" ]]; then
+          if ! git cat-file -e "${GIT_BASE_REF}" &> /dev/null; then
+            echo "::warning::Provided base reference ${GIT_BASE_REF} is invalid"
+            if [[ "${USE_FALLBACK}" == 'true' ]]; then
+              GIT_BASE_REF='HEAD~1'
+            fi
           fi
+        else
+          if ! git cat-file -e ${GITHUB_REF_BEFORE} &> /dev/null; then
+            GITHUB_REF_BEFORE='4b825dc642cb6eb9a060e54bf8d69288fbee4904'
+          fi
```

Comment:
I see that this was moved from the other actions, but what even is this ref?

Reply:
It's a "secret" git hash of an empty tree that is guaranteed to always exist.

Reply:
Oh. Neat.